src/wallet/wallet-generation.ts

```diff
@@ -0,0 +1,198 @@
+            address,
+            startingBalance
+          )
+
+          if (isAcctFunded) {
+            resolve(response.data)
+          } else {
+            reject(
+              new errors.XRPLFaucetError(
+                `Unable to fund address with faucet after waiting ${INTERVAL_SECONDS} seconds`
+              )
+            )
+          }
+        } catch (err) {
+          reject(new errors.XRPLFaucetError(err))
+        }
+      })
+      .catch((error) => {
+        reject(new errors.XRPLFaucetError(error.data))
+      })
+  })
+}
+
+/**
+ * Retrieves an XRPL address balance by currency
+ *
+ * @param api - RippleAPI
+ * @param address - Wallet address.
+ * @param currency - Currency to search, default XRP.
+ * @returns
+ */
+async function getWalletBalanceByCurrency(
+  api: RippleAPI,
+  address: string,
+  currency: string = 'XRP'
+): Promise<string> {
+  let curr = currency
+
+  // If issued currency longer than 3 characters
+  if (currency.length > 3) {
+    curr = Buffer.from(currency, 'ascii')
+      .toString('hex')
+      .toUpperCase()
+      .padEnd(40, '0')
+  }
+
+  try {
+    // Get all the account balances
+    const balances = await api.getBalances(address)
+
+    // Retrieve only the 'curr' (typically XRP) balance
+    const xrpCurrency = balances.filter((balance) => {
+      return balance.currency === curr
+    })
+
+    return xrpCurrency[0].value
+  } catch (err) {
+    reject(`Account ${address} does not exist`)
+  }
+}
+
+/**
+ * Check at regular interval if the account is enabled on the XRPL and funded
+ *
+ * @param api - RippleAPI
+ * @param address - the account address to check
+ * @param originalBalance - the initial balance before the funding
+ * @returns A Promise boolean
+ */
+async function hasAccountBalanceIncreased(
+  api: RippleAPI,
+  address: string,
+  originalBalance: number
+): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    let attempts = MAX_ATTEMPS
+    const interval = setInterval(async () => {
+      if (attempts < 0) {
+        clearInterval(interval)
+        reject(false)
+      } else {
+        attempts--
+      }
+
+      try {
+        const newBalance = +(await getWalletBalanceByCurrency(api, address))
+        if (newBalance > originalBalance) {
+          clearInterval(interval)
+          resolve(true)
+        }
+      } catch (err) {
+        reject(
+          new errors.XRPLFaucetError(
+            `Unable to check the account ${address} balance...`
+          )
+        )
+      }
+    }, INTERVAL_SECONDS * 1000)
+  })
+}
+
+/**
+ * Get the faucet URL based on the RippleAPI connection
+ * @param api - RippleAPI
+ * @returns A {@link FaucetNetwork}
+ */
+export function getFaucetUrl(api: RippleAPI) {
+  const connectionUrl = api.connection.getUrl()
+
+  // 'altnet' for Ripple Tesnet server and 'testnet' for XRPL Labs Testnet server
+  if (connectionUrl.includes('altnet') || connectionUrl.includes('testnet')) {
+    return FaucetNetwork.Testnet
```

Comment:
IMO this can just be constants, I don't think you really gain anything from having this be an `Enum` (but if this is good practice in JS then I'm happy to leave it alone)

Reply:
I agree. 
It's a bit easier to manipulate the faucet URLs with an `enum` because I use it in the tests as well.
And it's easier to have one `enum` import than two `const` import :)
it's a small thing but I think it's ok to leave the `enum` like that.